Makefile

```diff
@@ -258,14 +258,13 @@ uninstall:
 
 # find or download controller-gen
 # download controller-gen if necessary
-controller-gen:
-ifeq (, $(shell which controller-gen))
-	GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0
-	go mod tidy
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+controller-gen: __tooling-image
+CONTROLLER_GEN=docker run -it -v $(shell pwd):/gatekeeper gatekeeper-tooling controller-gen
+
+__tooling-image:
+	docker build . \
```

Comment:
We should figure out how to avoid rebuilding the docker image if it already exists.

Reply:
There's some tradeoffs here worth discussing:

1) `docker` will cache pre-built layers, though even cached it turns a 1.34s command into a 7.513s command

2) We can separate this into two discrete steps (build to an actual version tag like `1.0.0` instead of `latest`) and then reference this version in the other command.  This will result in speedup but will interrupt existing users' experience.  It will also inject additional complexity into the makefiles as there will need to be targets that a: build and run, b: run only.  The targets that use `controller-gen` as a dependency (`manifests`, `generate`) could then either have both "a" and "b" versions, or just always use "a".  If that's the case, then we're back where we started with "a" as the default.

Reply:
Max and I talked offline and decided we're going to keep this the way it is.